src/mutation.cpp

```diff
@@ -12,98 +12,308 @@
 #include <algorithm> //std::min
 #include <sstream>
 
-std::vector<std::string> unpowered_traits;
-
-bool Character::has_trait(const std::string &b) const
+bool Character::has_mut(const muttype_id &flag) const
 {
-    // Look for active mutations and traits
-    return my_mutations.find( b ) != my_mutations.end();
+    return my_mutations.find(flag) != my_mutations.end();
 }
 
-bool Character::has_base_trait(const std::string &b) const
+bool Character::is_trait(const muttype_id &flag) const
 {
-    // Look only at base traits
-    return my_traits.find( b ) != my_traits.end();
+    // Returns 1 or 0 depending on the existence of the trait
+    return my_traits.count(flag);
 }
 
-void Character::toggle_trait(const std::string &flag)
+bool Character::add_mutation(const muttype_id &flag, bool message)
 {
-    toggle_str_set(my_traits, flag); //Toggles a base trait on the player
-    toggle_str_set(my_mutations, flag); //Toggles corresponding trait in mutations list as well.
-    if( has_trait( flag ) ) {
+    // Only add the mutation if we don't already have it
+    if (!has_mut(flag)) {
```

Comment:
For the love of the space key: write `if( has_mut(...) ) { return false }` and remove one indentation level of the remaining code in that function.